ticket-proposal/tickets_idea_and_questions.md

```diff
@@ -0,0 +1,88 @@
+# Tickets Ideas and Questions
+
+## Design Considerations
+
+### Ticket Events
+Do we need an actual ticket created event?
+```
+{
+  "context": {
+    "version": "0.4.0-draft",
+    "id": "271069a8-fc18-44f1-b38f-9d70a1695819",
+    "source": "/event/source/123",
+    "type": "dev.cdevents.ticket.created.0.1.1",
+    "timestamp": "2023-03-20T14:27:05.315384Z"
+  },
+  "subject": {
+    // ticket related metadata
+  }
+}
+```
+
+How would an event know what ticket was associated with any CDEvent?
+
+So, maybe this approach would not work, since system would need to be aware of
+tickets which I do not know of any system that does this today. However, some
+standard need to be introduced to persist that ticketing metadata.
+
+Currently, how a lot of teams associate some ticket to a feature, is to
+mention it within a commit or the pull request.
+
+A git commit might look something like:
+```
+<ticket ID or link>: my feature that solves everything
+```
+
+Ticket system -> creates create_ticket event
+PR -> associates via commit that ticket link and/or create_ticket event -> source_change_created event -> passes this ticket metadata to all other events that are created
+
+```
+{
+  "context": {
+    "version": "0.4.0-draft",
+    "id": "271069a8-fc18-44f1-b38f-9d70a1695819",
+    "source": "/event/source/123",
+    "type": "dev.cdevents.ticket.linked.0.1.1",
+    "timestamp": "2023-03-20T14:27:05.315384Z"
+  },
+  "subject": {
+    // specifies the chain_id that it is apart of
+  }
+}
+```
+
+### Metadata on Existing CDEvents
```

Comment:
So it sounds like a hybrid approach, metadata on the actual CDEvents as well as separate ticketing events. My inclination when we started this assumed this would be the case, but I think we still need to figure out the benefits of having a separate event.